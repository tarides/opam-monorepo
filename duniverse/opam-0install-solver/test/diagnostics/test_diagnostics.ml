module Context = Opam_0install.Dir_context
module Solver = Opam_0install.Solver.Make(Context)

let main ~dir =
  let main_pkg = OpamPackage.Name.of_string "main" in
  let constraints = OpamPackage.Name.Map.empty in
  let env = Fun.const None in
  let context = Context.create ~constraints ~env dir in
  let result = Solver.solve context [main_pkg] in
  match result with
  | Ok selections ->
    List.iter
      (fun pkg -> print_endline (OpamPackage.to_string pkg))
      (Solver.packages_of_result selections)
  | Error diagnostics ->
    print_endline (Solver.diagnostics diagnostics)

let () =
  match Sys.argv with
  | [| _ ; dir |] -> main ~dir
  | _ -> assert false
