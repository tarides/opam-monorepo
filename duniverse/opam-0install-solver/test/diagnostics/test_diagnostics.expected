Can't find all required versions.
Selected: main.1
- foo -> (problem)
    main 1 requires >= 2
    Rejected candidates:
      foo.1: Incompatible with restriction: >= 2
      foo.2: Availability condition not satisfied
